### PR TITLE
ENH/DOC/TST/BLD: Clean up impala tests to enable parallel testing

### PR DESCRIPTION
**Pull request by cpcloud**
This adds support for running our entire test suite in parallel, reliably.

### CODE REVIEW

**Comment by kszucs** [comment]
Way more consistent this way. Should I review?

**Comment by cpcloud** [comment]
@kszucs can you review when you get a chance!

File: ci/impalamgr.py [hunk 1]
```diff
@@ -244,50 +244,61 @@ def load(data, udf, data_dir, overwrite):
...
-            print('Computing stats for', table.op().name)
+
+        def compute_stats(table):
+            logger.info('Computing stats for %s', table.op().name)
             table.compute_stats()
 
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            parquet_tables = create_parquet_tables(con, executor)
+            avro_tables = create_avro_tables(con, executor)
+            list(
```
Comment (kszucs) [inline ci/impalamgr.py]: It's a bit nested, decompose it into two expressions with a temporary variable?

File: ci/setup_docker_volume.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
Comment (kszucs) [inline ci/setup_docker_volume.sh]: Is this script used somewhere?

File: docs/source/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+   before continuing.
+
+.. _contrib.running_tests:
+
+Running the Test Suite
+----------------------
+
+Contributor `Krisztián Szűcs <https://github.com/kszucs>`_ has spent many hours
+crafting an easy-to-use `docker-compose <https://docs.docker.com/compose/>`_
+setup that enables ibis developers to get up and running quickly.
```
Comment (kszucs) [inline docs/source/contributing.rst]: Ohh, how nice! Thanks! :)

File: ci/setup_docker_volume.sh (hunk 2 above)
Comment (cpcloud) [reply ci/setup_docker_volume.sh]: My personal machine :)

File: ci/impalamgr.py (hunk 1 above)
Comment (cpcloud) [reply ci/impalamgr.py]: Sure

**Comment by cpcloud** [comment]
Merging on green.

**Pull request closed without merging**